public/app/plugins/panel/nodeGraph/layeredLayout.js

```diff
@@ -142,58 +136,41 @@ function createMappings(nodes, edges) {
   // Key is an iteration index and value is actual ID of the node
   const DOTToIdMap = {};
 
-  // Crate the maps both ways
   let index = 0;
-  for (const edge of edges) {
-    if (!idToDOTMap[edge.source]) {
-      idToDOTMap[edge.source] = index.toString(10);
-      DOTToIdMap[index.toString(10)] = edge.source;
-      index++;
-    }
+  for (const node of nodes) {
```

Comment:
Main fix is here, actually traversing the nodes instead of relying on every node being in the edges array.